internal/sync/provider_sync.go

```diff
@@ -26,17 +27,27 @@ import (
 	"sigs.k8s.io/cluster-api/util/conditions"
 
 	turtlesv1 "github.com/rancher-sandbox/rancher-turtles/api/v1alpha1"
+	"github.com/rancher-sandbox/rancher-turtles/internal/api"
 )
 
 // ProviderSync is a structure mirroring state of the CAPI Operator Provider object.
 type ProviderSync struct {
 	*DefaultSynchronizer
+	Destination api.Provider
 }
 
 // NewProviderSync creates a new mirror object.
 func NewProviderSync(cl client.Client, capiProvider *turtlesv1.CAPIProvider) Sync {
+	template := ProviderSync{}.Template(capiProvider)
+
+	destination, ok := template.(api.Provider)
+	if !ok {
+		return nil
```

Comment:
can we add a log entry here?

Reply:
This will never happen, annoyingly linter thinks that the `ok` value should be checked here. And the initialization is covered with tests, so this is proven.